src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -196,8 +196,7 @@ private BoundStatement BindFixedStatementParts(FixedStatementSyntax node, Diagno
 
         private BoundStatement BindYieldReturnStatement(YieldStatementSyntax node, DiagnosticBag diagnostics)
         {
-            var binder = GetBinder(node);
-            Debug.Assert(binder != null);
+            var binder = this;
```

Comment:
I left it like this intentionally in order to avoid unnecessary churn in the function.